snuba/web/query.py

```diff
@@ -47,6 +56,47 @@ def __init__(
         self.meta = meta
 
 
+@dataclass(frozen=True)
+class ClickhouseQueryMetadata:
+    sql: str
+    timer: Timer
+    stats: Mapping[str, Any]
+    status: str
+    trace_id: str
+
+    def to_dict(self):
+        return {
+            "sql": self.sql,
+            "timing": self.timer.for_json(),
+            "stats": self.stats,
+            "status": self.status,
+            "trace_id": self.trace_id,
+        }
+
+
+@dataclass
+class SnubaQueryMetadata:
+    request: Request
+    dataset: Dataset
+    timer: Timer
+    query_list: MutableSequence[ClickhouseQueryMetadata]
+    http_referrer: Optional[str] = ""
+
+    def to_dict(self):
+        return {
+            "http_referrer": self.http_referrer,
+            "dataset": str(self.dataset),
+            "query_list": [q.to_dict() for q in self.query_list],
```

Comment:
Mmmm, this is probably gonna record the same time on all the timers, will change this to record timings when we create the ClickhouseQueryMetadata